enhancements/security/openshift-image-policy.md

```diff
@@ -0,0 +1,338 @@
+approvers:
+  - "@mrunalp, machine-config and node staff engineer"
+api-approvers:
+  - "@JoelSpeed"
+creation-date: 2024-05-29
+last-updated: 2024-05-29
+tracking-link:
+  - https://issues.redhat.com/browse/OCPSTRAT-1245
+see-also:
+  - "/enhancements/api-review/add-ClusterImagePolicy-and-ImagePolicy-for-Signature-Verification.md"
+---
+
+# OpenShift Image Policy
+
+## Summary
+
+[An existing enhancement](/enhancements/api-review/add-ClusterImagePolicy-and-ImagePolicy-for-Signature-Verification.md) is adding ClusterImagePolicy and ImagePolicy primatives to config image policy.
+This enhancement describes how those primatives will be used by OpenShift clusters to secure core workloads:
+
+* The `quay.io/openshift-release-dev/ocp-release` scope, used for OpenShift release images, will require Sigstore signature signed by Red Hat, for any namespace in the cluster.
+* Users may add additional policies if they want, to add additional requirements as they see fit, including for the `quay.io/openshift-release-dev` scope.
+
+## Motivation
+
+As an OpenShift user, I want to verify the release image using [Sigstore](https://docs.sigstore.dev/about/overview/), so that I can verify those images come from Red Hat.
+
+### User Stories
+
+* As a cluster administrator, I want to be more confident that the OpenShift release image was signed by Red Hat, checking whenever a pod is launched from such an image, and not just at update-time.
+* As a cluster administrator, I want to be able to declare my own additional image policies, including restricting the use of the `quay.io/openshift-release-dev` scope.
+* As a Red Hat CI job maintainer or quality engineer, I have a way to disable Sigstore verification while testing releases before they have been signed.
+
+### Goals
+
+* Define an initial OpenShift ClusterImagePolicy that will be installed in all 4.17 clusters to restrict `quay.io/openshift-release-dev/ocp-release` to Red-Hat-signed images.
+
+### Non-Goals
+
+There are a number of possible next-steps that I'm punting on for now, as I try to nibble off the smallest possible slice of progress.
+This enhancement will not:
+
+* Figure out how to configure a ClusterImagePolicy for verifying a mirrored release image, although this will need to happen before we can use Sigstore signatures in disconnected/restricted-network environments.
+* Suggest changes to mirroring tools, although they will need to learn to mirror Sigstore signatures for use in disconnected/restricted-network environments.
+* Suggest changes to the ImageStream controller, although it will need to learn to mirror Sigstore signatures for verification when running pods from ImageStream sources, e.g. [must-gather][].
+* Suggest policies for `quay.io/openshift-release-dev/ocp-v4.0-art-dev`, although those would help tighten down security for other `openshift-*` namespaces.
+* Support testing unsigned releases as HyperShift hosted clusters on a management cluster that requires signed releases.
```

Comment:
As far as I'm aware, there is no signature verification today in the HyperShift stack.  In managed HyperShift (currently just hosted ROSA), ClusterImageSets define allowed update targets, and customers select from those presumably pre-verified targets.  The HyperShift rolls out whichever release pullspec without performing any signature verification of its own, although it does run some pre-checks like [looking at `Upgradeable`][1].

[1]: https://github.com/openshift/hypershift/blame/99ca57bddd3280537bbb5c04ffe711e13c823a9c/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L4718-L4765

Reply:
If someone can patch `HostedCluster` and roll out an untrusted payload, I think we must close this gap.

Reply:
Makes sense to me, but:

* Do you want to close that gap via OpenPGP / GPG signatures?  Because if you do, that's a completely separate effort from this enhancement's Sigstore proposal.
* If you are comfortable closing that gap via management clusters having a ClusterImagePolicy per this proposal, the gap will close, although without some HyperShift-side investigation, the user experience of failing to launch Sigstore-troubled release-image pods is unclear.  But closing that gap will take time, as this enhancement percolates through tech-preview to GA, and management clusters eventually update to pick up the GA version.

Reply:
> closing that gap via management clusters having a ClusterImagePolicy per this proposal

A corner case might be the management cluster enforcing a production-key-signed policy, while trying to roll out a beta-key-signed prerelease of the managed cluster. Can that happen at all?